src/app/public/modules/restricted-view/restricted-view.component.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Component,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  SkyRestrictedViewAuthService
+} from './restricted-view-auth.service';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+@Component({
+  selector: 'sky-restricted-view',
+  templateUrl: './restricted-view.component.html'
+})
+export class SkyRestrictedViewComponent implements OnDestroy {
+  public isAuthenticated = false;
+
+  private ngUnsubscribe = new Subject<void>();
+
+  constructor(
+    private authService: SkyRestrictedViewAuthService
+  ) {
+    this.authService.isAuthenticated
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((isAuthenticated: boolean) => {
+        this.isAuthenticated = isAuthenticated;
```

Comment:
We can use Angular's `AsyncPipe` to make this logic _much_ simpler.

```
public isAuthenticated: Observable<boolean>;

constructor(
  private authService: SkyRestrictedViewAuthService
) {
  this.isAuthenticated = this.authService.isAuthenticated;
}
```

Then, in your HTML template:

```
<ng-content *ngIf="isAuthenticated | async"></ng-content>
```

The `AsyncPipe` automatically unsubscribes from the observable, too, so you won't need the `ngUnsubscribe` subject.